### PR TITLE
feat: add schema injection

### LINKED ISSUE #4591: Injection of Schema Selections into Flux Editor
## User Story

<p></p>

As a user, I want my flux script started for me by using the metrics that I have chosen in the schema browser, so that I don't have to type as many lines of code.

<p></p>

## Solution Overview

<p></p>

The schema browser (at the end of the basic schema browser user story) allows the user to select a bucket and measurement and shows them a list of fields, tags, and tag values.  This story adds the ability to hover over a field, tag, or tag value and select an "Inject" button that will appear.  

<p></p>

## Acceptance Criteria

<p></p>

- When a user hovers over a field, tag, or tag value in schema browser, an "inject" button will appear

- Upon clicking on the inject button, the software will formulate the appropriate flux script lines per the following rules:

  - If no current from/range/measurement filter exists in the flux script then a from range statement will be added using the bucket that is currently selected and a filter statement will be added for the currently selected measurement

  - If a from/range statement already exists, but no measurement in the script, then a measurement filter will be added

- When the object being injected is a field, the software fill formulate the appropriate flux script line  per the following rules:

  - If a filter line exists already for a field, the injected field will be added to the existing flux filter line for that field with an "OR" and the new field name.

  - If a filter line does not already exist for a field in this search, then a filter line will be added for this field.

- When the object being injected is a Tag, the software fill formulate the appropriate flux script line  per the following rules:

  - If a filter line exists already for a tag, the injected tag will be added to the existing flux filter line for tags with an "OR" and the new tag name.

  - If a filter line does not already exist for a tag in this search, then a filter line will be added for this tag.

- When the object being injected is a Tag value, the software fill formulate the appropriate flux script line  per the following rules:

  - If a filter line exists already for a tag value, the injected tag value will be added to the existing flux filter line for tags with an "AND" and the new tag value name.

  - If a filter line does not already exist for a tag value in this search, then a filter line will be added for this tag value.

- If a user changes bucket or measurement when there is already a from range and filter for that bucket/measurement and attempts to do an inject of a field, tag, or tag value - then the user will be shown a dialog that asks them whether they want to replace the existing script or append to the existing script and based upon that selection the software will either replace or append.

- When a from range filter is injected into the script, a pivot and yield flux statement will also be inserted (need to discuss with QX team)

<p></p>

## Notes

<p></p>

- This A/C needs reviewed by QE team for accuracy.

- The Replace/Append item needs UI/UX Design

<p></p>

### PR DESCRIPTION
**Pull request by appletreeisyellow**
Closes #4591 

This PR adds the front-end part of the schema injection in MVP version. 

## After

https://user-images.githubusercontent.com/14298407/176258689-06ff42e8-4a4e-4048-b347-b0cbfc030542.mov

### CODE REVIEW

File: src/dataExplorer/context/fluxQueryBuilder.tsx [hunk 1]
```diff
@@ -95,6 +124,39 @@ export const FluxQueryBuilderProvider: FC = ({children}) => {
     getTagKeys(selection.bucket, measurement)
   }
 
+  const handleSelectField = (field: string): void => {
+    // Inject field
+    injectViaLsp(ExecuteCommand.InjectField, {
+      bucket:
```
Comment (drdelambre) [inline src/dataExplorer/context/fluxQueryBuilder.tsx:130]: why do we need the bucket here? isn't that just used on measurement injection for the current iteration?

File: src/dataExplorer/context/fluxQueryBuilder.tsx [hunk 2]
```diff
@@ -95,6 +124,39 @@ export const FluxQueryBuilderProvider: FC = ({children}) => {
...
+    event('flux.schema.injected', {
+      command: ExecuteCommand.InjectField,
+      context: 'flux query builder',
+    })
+  }
+
+  const handleSelectTagValue = (tagKey: string, tagValue: string): void => {
+    // Inject tag value
+    injectViaLsp(ExecuteCommand.InjectTagValue, {
+      bucket:
```
Comment (drdelambre) [inline src/dataExplorer/context/fluxQueryBuilder.tsx:146]: same question about buckets here

File: src/dataExplorer/context/fluxQueryBuilder.tsx (hunk 1 above)
Comment (appletreeisyellow) [reply src/dataExplorer/context/fluxQueryBuilder.tsx:130]: It's required in LSP https://github.com/influxdata/flux-lsp/blob/master/src/server/mod.rs#L1333-L1336

Here is my understanding:

No matter the user is injecting measurement, field, or tag value in UI, LSP treats them as injecting `filter()` function. In order to make sure the injection line works, LSP asks for the bucket that filter function is applying to. 

Or there are other intentions? Correct me if I understand it wrong @rockstar

**Review by wiedld — APPROVED** [review]

File: src/dataExplorer/context/fluxQueryBuilder.tsx (hunk 1 above)
Comment (wiedld) [reply src/dataExplorer/context/fluxQueryBuilder.tsx:130]: Append only, to the last flux pipeline in the query.
Confirms that the last flux transformation pipeline has the same bucket as we are seeking.

This is the first iteration of the injection in the LSP.
Plans are already in the works for the next one.

**Review by drdelambre — APPROVED** [review]

**Pull request merged**